library/objective-c/EnvoyEngineImpl.m

```diff
@@ -365,12 +365,26 @@ static envoy_data ios_get_string(const void *context) {
   return toManagedNativeString(accessor.getEnvoyString());
 }
 
+static void ios_track_event(envoy_map map, const void *context) {
+  // This code block runs inside the Envoy event loop. Therefore, an explicit autoreleasepool block
+  // is necessary to act as a breaker for any Objective-C allocation that happens.
+  @autoreleasepool {
+    EnvoyEventTracker *eventTracker = (__bridge EnvoyEventTracker *)context;
+    // release memory even if the tracking closure is not provided.
```

Comment:
Is this docstring relevant to a specific line of code? I don't see an explicit release here

Reply:
the release happens inside of `to_ios_map` method in [here](https://github.com/envoyproxy/envoy-mobile/pull/1590/files#diff-271bc2a427a66a65657bc509a808ff32e2536938f9a6365a8731e5ed50d4b9adR116). Not  obvious but it seems to be following a pattern established by other methods such as `to_ios_headers`. @goaway maybe we should rename this method to make their memory-related behavior more obvious? `bridge_to_ios_map` is an alternate name that comes to my mind here.